loopchain/blockchain/blockchain.py

```diff
@@ -237,6 +237,21 @@ def find_confirm_info_by_height(self, height) -> bytes:
 
         return bytes()
 
+    def find_preps_by_roothash(self, roothash: str) -> bytes:
+        hash_encoded = roothash.encode(encoding='UTF-8')
+        try:
+            return bytes(self._blockchain_store.get(BlockChain.PREPS_KEY + hash_encoded))
+        except KeyError:
+            return bytes()
+
+    def save_preps(self, roothash: str, preps: dict, batch=None):
```

Comment:
I think `batch` argument could raise misunderstanding.
How about add type hint `KeyValueStoreWriteBatch` or something..?

```suggestion
    def save_preps(self, roothash: str, preps: dict, batch: KeyValueStoreWriteBatch = None):
```